presto-main/src/main/java/com/facebook/presto/cost/SimpleFilterProjectSemiJoinStatsRule.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.plan.FilterNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.ProjectNode;
+import com.facebook.presto.sql.planner.plan.SemiJoinNode;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.NotExpression;
+import com.facebook.presto.sql.tree.SymbolReference;
+import com.google.common.collect.Iterables;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.sql.ExpressionUtils.combineConjuncts;
+import static com.facebook.presto.sql.ExpressionUtils.extractConjuncts;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class SimpleFilterProjectSemiJoinStatsRule
+        implements ComposableStatsCalculator.Rule
+{
+    private static final Pattern PATTERN = Pattern.matchByClass(FilterNode.class);
+
+    private final FilterStatsCalculator filterStatsCalculator;
+    private final SemiJoinStatsCalculator semiJoinStatsCalculator;
+
+    public SimpleFilterProjectSemiJoinStatsRule(FilterStatsCalculator filterStatsCalculator, SemiJoinStatsCalculator semiJoinStatsCalculator)
+    {
+        this.filterStatsCalculator = requireNonNull(filterStatsCalculator, "filterStatsCalculator can not be null");
+        this.semiJoinStatsCalculator = requireNonNull(semiJoinStatsCalculator, "semiJoinStatsCalculator can not be null");
+    }
+
+    @Override
+    public Pattern getPattern()
+    {
+        return PATTERN;
+    }
+
+    @Override
+    public Optional<PlanNodeStatsEstimate> calculate(PlanNode node, Lookup lookup, Session session, Map<Symbol, Type> types)
+    {
+        FilterNode filterNode = (FilterNode) node;
+        SemiJoinNode semiJoinNode;
+        if (filterNode.getSource() instanceof ProjectNode) {
+            ProjectNode projectNode = (ProjectNode) filterNode.getSource();
+            if (!projectNode.isIdentity()) {
```

Comment:
You don't need `projectNode` to be identify. You need only that it's identity for the symbol produced by `SemiJoinNode`